src/main/java/com/purple/ua/universityappointment/service/impl/AppointmentServiceImpl.java

```diff
@@ -172,18 +146,17 @@ public AppointmentDto updateStatus(long id, Status status) {
                 () -> new AppointmentNotFoundException(HttpStatus.NOT_FOUND, "Appointment with id: " + id + " not found"));
         appointment.setStatus(status);
         Appointment save = appointmentRepository.save(appointment);
-        mailInput(save.getStudent().getEmail(), "Appointment status!",
+        mailUtils.mailInput(save.getStudent().getEmail(), "Appointment status!",
                 "Your appointment for " + save.getLesson().getFromDate() + " is: " + save.getStatus());
         return INSTANCE.toDto(save);
     }
 
-
-    public void mailInput(String email, String subject, String text)  {
-        SimpleMailMessage mailMessage = new SimpleMailMessage();
-        mailMessage.setTo(email);
-        mailMessage.setSubject(subject);
-        mailMessage.setText(text);
-        emailSenderService.sendEmail(mailMessage);
+    public void timeCheck(Lesson lesson, LocalDateTime startTime, LocalDateTime endTime) {
+        if (!endTime.isAfter(startTime)) {
+            throw new TimeConflictException(HttpStatus.CONFLICT, "DateFrom is after ToDate");
+        }
+        if ((startTime.isBefore(lesson.getFromDate()) || endTime.isAfter(lesson.getToDate()))) {
+            throw new TimeConflictException(HttpStatus.CONFLICT, "The date doesn't match available time");
+        }
```

Comment:
rename to ckeckTime, due to Java Code Convention method name should start from verb